### PR TITLE
Add dedent pkg

### PR DESCRIPTION
**Pull request by ghost**
[Analytics]()

PR for adding dedent package to the kubernetes. 
Used to test latest godeps documentation. Next PR will use dedent package in kubectl's describe command. 

cc @janetkuo 
cc @thockin 
See  #17609

### CODE REVIEW

**Comment by mhrgoog** [comment]
@k8s-bot e2e test this issue: #25812

**Comment by thockin** [comment]
as discussed you want 2 commits in the same PR.  one for the dep, the other for the consumer.

**Comment by mhrgoog** [comment]
This pull request now has the two commits as discussed.

**Comment by mhrgoog** [comment]
@k8s-bot unit test this issue: #IGNORE

We are kick starting this again as it failed before since go4.org has an expired TLS

File: pkg/kubectl/cmd/describe.go [hunk 1]
```diff
@@ -39,36 +40,39 @@ type DescribeOptions struct {
 	Recursive bool
 }
 
-const (
-	describe_long = `Show details of a specific resource or group of resources.
+var (
+	describe_long = dedent.Dedent(`
+	Show details of a specific resource or group of resources.
```
Comment (thockin) [inline pkg/kubectl/cmd/describe.go]: if we are nit-picking I would indent this one extra level.  But it's minor and not worth re-piushing unless there are other things to change

**Comment by thockin** [comment]
so the real fun is all of the other kubectl per-command files that follow the same pattern :)  This was just one of them.

**Comment by pwittrock** [comment]
@mhrgoog Looks good.  Another rebase is required.  Probably another update to GoDeps.  Rebases will continue to be a problem as long as this is in the queue, so we may want to bump it to the front once all of the critical stuff for 1.3 is in.  I am not sure how we are planning on implementing the feature freeze for 1.3

File: pkg/kubectl/cmd/describe.go (hunk 1 above)
Comment (mhrgoog) [inline pkg/kubectl/cmd/describe.go]: done. I had to rebase anyway.

**Comment by janetkuo** [comment]
Is this PR planning to cover all commands? Some commands are missing, e.g. `kubectl rollout`.

**Comment by janetkuo** [comment]
We should document the use of dedent package, at least here https://github.com/kubernetes/kubernetes/blob/master/docs/devel/kubectl-conventions.md#command-implementation-conventions

**Comment by janetkuo** [comment]
The 2nd and 3rd commits can be squashed into one.

**Comment by mhrgoog** [comment]
I have one more commit coming that should have all the commands (including rollout and config and others) using dedent. 

Also I will squash commit two and three and the new one all into one commit.

I will update the documentation in one final commit.

Let me know if that does not sound ok.

File: docs/devel/kubectl-conventions.md [hunk 2]
```diff
@@ -279,15 +279,17 @@ type MineConfig struct {
...
+var (
+  mineLong = dedent.Dedent(`
+	Some long description
+	for my command.`)
 
-  mineExample = `  # Run my command's first action
-  $ %[1]s first
+  mineExample = dedent.Dedent(`
+	  # Run my command's first action
+	  $ %[1]s first
```
Comment (janetkuo) [inline docs/devel/kubectl-conventions.md]: Can you remove the `$` sign also?

Also `%[1]s first` looks mysterious to me. Can you change it to something like `kubectl <command> <first_action>`

Comment (mhrgoog) [inline docs/devel/kubectl-conventions.md]: Done. Thanks.

**Comment by janetkuo** [comment]
:+1: LGTM now

**Comment by pwittrock** [comment]
@k8s-bot unit test this issue: #25363 

[artifacts](https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/25922/kubernetes-pull-test-unit-integration/27769/)

**Comment by pwittrock** [comment]
lgtm

**Comment by mhrgoog** [comment]
Adding lgtm as I just reran hack/save-godep.sh

**Pull request closed without merging**